content/architecture/access-control.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Access Control
+linktitle: Access Control
+aliases: [/rbac/]
+description: Managing Access Control
+date: 2017-02-01
+publishdate: 2017-02-01
+lastmod: 2017-02-01
+menu:
+  docs:
+    parent: "architecture"
+    weight: 101
+weight: 110
+sections_weight: 101
+draft: false
+toc: true
+---
+
+Jenkins X uses a Role Based Access Control (RBAC) policies to control access to its various resources.  The enforcement of the policies is provided by [Kubernetes' RBAC support](https://kubernetes.io/docs/reference/access-authn-authz/rbac/).
+
+As a given [Team](/about/features/#teams) can have a number of [Environments](/about/features/#environments) (e.g. Dev, Staging, Production) along with dynamic [Preview Environments](/developing/preview/) it can be a challenge keeping all the [`Role`](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) and [`RoleBinding`](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) resources from [Kubernetes RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) in sync with all the various namespaces and members of your team.
+
+In order to make this management easier Jenkins X creates a new [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) called [`EnvironmentRoleBinding`](architecture/custom-resources/#environmentrolebinding) which allows you to associate a `Role` labelled with `jenkins.io/kind=EnvironmentRole` with as many `Users` or `ServiceAccounts` as you like. Then as Environments are created or the `Role` or `EnvironmentRoleBinding` in the Dev environment is modified, the [`role controller`](/commands/jx_controller_role/#jx-controller-role) ensures that the configuration is replicated to all the environment namespaces by creating or updating all of the `Role` and `RoleBinding`s per namespace
+
+Roles are per Team so it is possible to have special roles per team, or to use common names for roles but have them tweaked on a per team basis.
+
+## Security Implications for the admin namespace
+
+Jenkins X stores various configuration / settings (e.g. `Users`, `Teams`) in the main admin namespace (jx), therfore care should be taken when granting roles in the default `jx` team as allowing users to edit some of these files may allow them to escalate their permissions.
+It is therefor not recommended to grant non admin users access to this namespace but to create teams and grant users access to those when using a shared cluster.
```

Comment:
```suggestion
Instead of granting non-admin users access to the `jx` namespace, create teams and grant users access to those when using a shared cluster.
```